scaaml/io/resume_kti.py

```diff
@@ -11,15 +11,17 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-"""Resumable iterator for key-text pairs with autosaves.
+"""Resumable iterator for key-text pairs (or more general tuples) with
+autosaves.
 
 Autosaves after shard length iterations.
 
 Typical usage example:
 
   # Save all key-text pairs ahead before the experiment starts.
   # If kt_filename exists this does nothing.
-  create_resume_kti(keys, texts, shard_length=shard_length,
+  create_resume_kti(parameters={"keys": keys, "texts": texts},
+                    shard_length=shard_length,
                     kt_filename="key_text_pairs.txt",
```

Comment:
Technically it's not a pair anymore if the dict has more than 2 keys. How about renaming these to parameters_tuples.txt and progress_tuples.txt?